src/app/courses/pages/courses/courses.component.ts

```diff
@@ -54,28 +53,78 @@ export class CoursesComponent implements OnInit {
     this.loadMore();
   }
 
+  canAddCourse(): Boolean {
+    return this.authService.isAuthenticated();
+  }
+
+  canEditCourse(course: Course): Boolean {
+    // TODO add check right to owner Course here
+
+    return this.authService.isAuthenticated();
+  }
+
+  canDeleteCourse(course: Course): Boolean {
+    // TODO add check right to owner Course here
+
+    return this.authService.isAuthenticated();
+  }
+
   addCourse() {
     let course = new Course();
-    this.dialogAddCourse = this.dialog.open(AddCourseComponent, {
+    let dialogRef = this.dialog.open(AddCourseComponent, {
       data: { course },
       disableClose: false,
     });
+
+      dialogRef.componentInstance.onSave.subscribe(course => {
+        // console.log(course);
+        try {
+          this.coursesService.add(course).subscribe(result => {
```

Comment:
Also you should add this new course in your course list. (Like deleting course from the list when `delete(course)` was called)
Or some another logic to show new course